docs/src/components/SearchInput/SearchInput.component.tsx

```diff
@@ -0,0 +1,137 @@
+    id: string;
+    url?: string;
+};
+
+interface InitialState {
+    client?: any;
+    index?: object | boolean;
+    dataSource?: [DataSet];
+}
+
+export class SearchInput extends React.PureComponent<InitialState> {
+    readonly state = {
+        client: undefined,
+        index: {},
+        dataSource: [],
+    };
+
+    constructor(props: object) {
+        super(props);
+        this.searchIndex = debounce(
+            this.searchIndex.bind(this),
+            DEBOUNCE_DELAY
+        );
+    }
+
+    componentDidMount() {
+        // Initializes Algolia's API and assigns it to state
+        const client = algoliasearch(APP_ID, API_KEY);
+        const index = client.initIndex(INDEX_NAME);
+
+        this.setState({
+            client,
+            index,
+        });
+    }
+
+    // Searches the Algolia index and formats the results for the Autocomplete component
+    async searchIndex(val: string) {
+        const { index } = this.state;
+        const dataSource: DataSet[] = [];
+
+        /*
+            Searches Algolia index for results.
+            TODO: Typescript error based on the fact that index doesn't have the search property
+            initially. Gating it doesn't fix the issue.
+        */
+        const results = await index.search({
+            query: val,
+            hitsPerPage: NUMBER_OF_RESULTS,
+        });
+
+        if (results) {
+            // 'hits' is an Algolia term for matching results
+            const { hits } = results;
+            let previousSection = '';
+
+            // TODO: This probably could be converted to use .reduce, but moving quickly
+            hits.forEach((hit: AlgoliaHit) => {
+                /*
+                    Algolia's scraper ranks search terms/sections in a hierarchy (lvl0-lvl5) lvl0
+                    is higher up in the page tree, i.e. COMPONENTS, whereas lvl5 might be a code
+                    sample title. This extracts those values to make the search results for
+                    Autocomplete.
+                */
+                const hitValues = Object.values(hit.hierarchy);
+                // This is the main section of the site and will be a title not a linked result.
+                const hitSection = hitValues.shift();
+                // Joins the rest of the hit hierarchy values, filtering out nulls.
+                const hitLabel = hitValues
+                    .filter((value: string) => value !== null)
+                    .join(' > ');
+
+                /*
+                    Don't want titles repeated if they're right next to each other. The order of
+                    Algolia's results appear to be based on relevance, so it's possible that further
+                    down the list the same title will appear.
+                */
+                if (hitSection !== previousSection) {
+                    dataSource.push({
+                        label: hitSection,
+                        id: `${hit.objectID}-title`,
```

Comment:
If you press the down arrow key to go tab through the results, the titles (e.g. `COMPONENTS`) are included in the autocomplete selections. If you add `listItemType: 'title'` here, it'll solve the issue.

Reply:
Your `Results` template is being used, but the `Autocomplete` uses the `dataSource` to determine which items should be selectable/tabbable, so you still need to specify the `listItemType`